src/sage/geometry/lattice_polytope.py

```diff
@@ -309,12 +283,13 @@ def LatticePolytope(data, compute_vertices=True, n=0, lattice=None):
         sage: p.faces()                                                                 # needs sage.graphs
         ((-1-d lattice polytope in 3-d lattice M,),)
     """
+    parent = LatticePolytopes()
     if isinstance(data, LatticePolytopeClass):
         data = data._vertices
         compute_vertices = False
     if (isinstance(data, PointCollection) and
-        (lattice is None or lattice is data.module())):
-        return LatticePolytopeClass(data, compute_vertices)
+            (lattice is None or lattice is data.module())):
+        return LatticePolytopeClass(parent, data, compute_vertices)
```

Comment:
```suggestion
        return parent.element_class(parent, data, compute_vertices)
```
This should fix the TestSuite category check failure.